code/obj/machinery.dm

```diff
@@ -368,6 +368,6 @@
 	var/area/A1 = get_area(src)
 	. = ..()
 	var/area/A2 = get_area(src)
-	if(A1 != A2)
+	if(islist(A1?.machines) && islist(A2?.machines) && A1 != A2)
```

Comment:
why are you checking islist instead of just nullchecks

Reply:
I could probably do null-checks here instead of list-checks